File: tools/reference/longest-palindromic-substring.js
var longestPalindrome = function (s) {
  let best = '';
  const grow = (lo, hi) => {
    while (lo >= 0 && hi < s.length && s[lo] === s[hi]) {
      lo--;
      hi++;
    }
    const candidate = s.slice(lo + 1, hi);
    if (candidate.length > best.length) best = candidate;
  };
  for (let i = 0; i < s.length; i++) {
    grow(i, i);
    grow(i, i + 1);
  }
  return best;
};

File: tools/reference/longest-palindromic-substring.py
def longestPalindrome(s):
    best = ''
    for center in range(len(s)):
        for lo, hi in ((center, center), (center, center + 1)):
            while lo >= 0 and hi < len(s) and s[lo] == s[hi]:
                lo -= 1
                hi += 1
            candidate = s[lo + 1:hi]
            if len(candidate) > len(best):
                best = candidate
    return best
